packages/ButtonGroup/__tests__/ButtonGroup.spec.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+
+import A11yContent from '@tds/core-a11y-content'
+
+import ButtonGroup from '../ButtonGroup'
+
+describe('ButtonGroup', () => {
+  const defaultProps = {
+    name: 'storageSize',
+    onChange: () => {
+      return true
+    },
+    value: '64gb',
+    label: 'Please select a storage size',
+  }
+  const doShallow = (props = {}) =>
+    shallow(
+      <ButtonGroup {...props}>
+        <ButtonGroup.Item value="64gb">64 GB</ButtonGroup.Item>
+        <ButtonGroup.Item value="128gb">128 GB</ButtonGroup.Item>
+        <ButtonGroup.Item value="256gb">256 GB</ButtonGroup.Item>
+      </ButtonGroup>
+    )
+
+  const doMount = (props = {}) =>
+    mount(
+      <ButtonGroup {...props}>
+        <ButtonGroup.Item value="64gb">64 GB</ButtonGroup.Item>
+        <ButtonGroup.Item value="128gb">128 GB</ButtonGroup.Item>
+        <ButtonGroup.Item value="256gb">256 GB</ButtonGroup.Item>
+      </ButtonGroup>
+    )
+
+  it('renders', () => {
+    const buttonGroup = doShallow(defaultProps)
+
+    expect(buttonGroup).toMatchSnapshot()
+  })
+
+  it('selects the correct button', () => {
+    const buttonGroup = doMount(defaultProps)
+    expect(
+      buttonGroup
+        .children()
+        .find('input')
```

Comment:
This use of `.find` finds the first matching element, which happens to be selected button in this test. This test would fail if `value` were to equal `128gb`, even though it may actually be properly selected.

Try:

```jsx
const buttonGroup = doMount({ value="128gb" }) // assumes `defaultProps` are passed

expect(
  buttonGroup
    .children()
    .find('input[value="128gb"]')
    .get(0).props.checked
).toEqual(true)

```